src/AdvApi32.Desktop/AdvApi32.Helpers.cs

```diff
@@ -47,36 +178,46 @@ public static void InstallService(string servicePath, string serviceName, string
                     {
-                        lpDescription = serviceDescription
+                        lpDescription = lpDescription
                     };
 
                     IntPtr lpInfo = Marshal.AllocHGlobal(Marshal.SizeOf(descriptionStruct));
 
-                    if (lpInfo == IntPtr.Zero)
+                    try
                     {
-                        throw new Win32Exception();
+                        Marshal.StructureToPtr(descriptionStruct, lpInfo, false);
+                        if (!ChangeServiceConfig2(svcHandle, ServiceInfoLevel.SERVICE_CONFIG_DESCRIPTION, lpInfo))
+                        {
+                            throw new Win32Exception();
+                        }
                     }
-
-                    Marshal.StructureToPtr(descriptionStruct, lpInfo, false);
-
-                    if (!ChangeServiceConfig2(svcHandle, ServiceInfoLevel.SERVICE_CONFIG_DESCRIPTION, lpInfo))
+                    finally
                     {
+                        Marshal.DestroyStructure(lpInfo, typeof(ServiceDescription));
                         Marshal.FreeHGlobal(lpInfo);
-                        throw new Win32Exception();
-                    }
-
-                    Marshal.FreeHGlobal(lpInfo);
-
-                    if (svcHandle.IsInvalid)
-                    {
-                        throw new Win32Exception();
                     }
                 }
             }
         }
 
-        public static void UninstallService(string serviceName)
+        /// <summary>
+        /// Marks the specified service for deletion from the service control manager database on the local computer.
+        /// </summary>
+        /// <param name="lpServiceName">
+        /// The name of the service to be opened. This is the name specified by the lpServiceName parameter of the <see cref="CreateService(SafeServiceHandle,string,string,ServiceAccess,ServiceType,ServiceStartType,ServiceErrorControl,string,string,int, string,string,string)"/> function when the service object was created,
+        /// not the service display name that is shown by user interface applications to identify the service.
+        /// The maximum string length is 256 characters. The service control manager database preserves the case of the characters,
+        /// but service name comparisons are always case insensitive. Forward-slash (/) and backslash (\) are invalid service name characters.
+        /// </param>
+        /// <exception cref="Win32Exception">If the method fails, returning the calling thread's last-error code value.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="lpServiceName" /> is NULL or an empty string.</exception>
+        public static void DeleteService(string lpServiceName)
         {
+            if (string.IsNullOrWhiteSpace(lpServiceName))
+            {
+                throw new ArgumentNullException(nameof(lpServiceName));
```

Comment:
I'll go ahead and merge since you've done a lot of good work here and this is a net positive. But please consider following up with another short change here.